src/Events/Category_Colors/Controller.php

```diff
@@ -13,9 +29,7 @@ class Controller extends Controller_Contract {
 	 * @since TBD
 	 */
 	public function do_register(): void {
-		$this->container->singleton( Category_Colors::class );
-		$this->container->singleton( Category_Colors_Settings::class );
-		$this->container->singleton( Quick_Edit::class );
+		$this->container->bind( Event_Category_Meta::class );
```

Comment:
This isn't a correct binding - the container wont be able to decide how to build a class which requires an int in its constructor.

You have no way of binding an implementation that requires a int which could change every time in your container.